### PR TITLE
JSDoc for src/*/*

### PR DESCRIPTION
**Pull request by Computroniks**
# Description

Related to https://github.com/louislam/uptime-kuma/pull/1499

This PR adds JSDoc comments for the `src/*/*` directory. For the full TODO list, see https://github.com/louislam/uptime-kuma/pull/1499

## Type of change

Please delete any options that are not relevant.

- Other

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I ran ESLint and other linters for modified files
- [ ] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [ ] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

### CODE REVIEW

**Comment by Computroniks** [comment]
Sorry for the delays with this. I am currently doing exams at school so I will try get this done as and when I have time.

File: src/components/TagsManager.vue [hunk 1]
```diff
@@ -307,26 +323,51 @@ export default {
...
+         * @param {number} monitorId ID of monitor to remove tag from
+         * @param {string} value Value of tag
+         * @returns {Promise<void>}
+         */
         deleteMonitorTagAsync(tagId, monitorId, value) {
             return new Promise((resolve) => {
                 this.$root.getSocket().emit("deleteMonitorTag", tagId, monitorId, value, resolve);
             });
         },
+        /** Called as user types input */
```
Comment (chakflying) [inline src/components/TagsManager.vue:360]:
```suggestion
        /** Handle pressing Enter key when inside the modal */
```

File: src/components/ToggleSection.vue [hunk 2]
```diff
@@ -29,10 +29,12 @@
 <script>
 export default {
     props: {
+        /** Heading to use */
```
Comment (chakflying) [inline src/components/ToggleSection.vue:32]:
```suggestion
        /** Heading of the Section */
```

File: src/components/ToggleSection.vue [hunk 3]
```diff
@@ -29,10 +29,12 @@
...
         heading: {
             type: String,
             default: "",
         },
+        /** Should the selection be open by default? */
```
Comment (chakflying) [inline src/components/ToggleSection.vue:37]:
```suggestion
        /** Should the section be open by default? */
```

**Review by chakflying — CHANGES_REQUESTED** [review]

File: src/mixins/mobile.js [hunk 4]
```diff
@@ -12,11 +12,13 @@ export default {
     },
 
     methods: {
+        /** Called when the screen changes size */
```
Comment (Computroniks) [inline src/mixins/mobile.js:15]:
```suggestion
        /** Handle screen resize */
```

File: src/pages/Settings.vue [hunk 5]
```diff
@@ -118,13 +118,17 @@ export default {
 
     methods: {
 
-        // For desktop only, mobile do nothing
+        /**
+         * Load the general settings page
+         * For desktop only, mobile do nothing
```
Comment (Computroniks) [inline src/pages/Settings.vue:123]:
```suggestion
         * For desktop only, on mobile do nothing
```

File: src/components/Confirm.vue [hunk 6]
```diff
@@ -50,9 +53,13 @@ export default {
         this.modal = new Modal(this.$refs.modal);
     },
     methods: {
+        /** Show the confirm dialog */
         show() {
             this.modal.show();
         },
+        /**
+         * @emits "yes" - Notify the parent when Yes is pressed
```
Comment (Computroniks) [inline src/components/Confirm.vue:61]:
```suggestion
         * @emits string "yes" Notify the parent when Yes is pressed
```

**Review by Computroniks — COMMENTED** [review]
Just been through and reviewed the comments to check for any silly errors. I think that this should be ready to be converted to active

**Review by chakflying — CHANGES_REQUESTED** [review]

**Pull request merged**